main.tf

```diff
@@ -89,11 +89,16 @@ resource "google_storage_bucket_object" "main" {
 }
 
 // todo(bharathkkb): remove workaround after https://github.com/hashicorp/terraform-provider-google/issues/11383
+// Also: https://github.com/hashicorp/terraform/issues/28925 (when this functions project is created)
 data "google_project" "nums" {
-  for_each   = toset(concat(compact([for item in var.secret_environment_variables : lookup(item, "project_id", "")]), [var.project_id]))
+  for_each   = toset(compact([for item in var.secret_environment_variables : lookup(item, "project_id", "")]))
   project_id = each.value
 }
 
+data "google_project" "main" {
```

Comment:
```suggestion
data "google_project" "default" {
```

Reply:
Followed your suggestion to rename `main` to `default`